src/backend/distributed/utils/multi_partitioning_utils.c

```diff
@@ -130,6 +138,107 @@ worker_fix_pre_citus10_partitioned_table_constraint_names(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * fix_partition_shard_index_names fixes the index names of shards of partitions of
+ * partitioned tables on workers.
+ *
+ * When running CREATE INDEX on parent_table, we didn't explicitly create the index on
+ * each partition as well. Postgres created indexes for partitions in the coordinator,
+ * and also in the workers. Actually, Postgres auto-generates index names when auto-creating
+ * indexes on each partition shard of the parent shards. If index name is too long, it
```

Comment:
The current comment is correct as I don't mention that PG autogenerates only for long names. I am however explaining the issue in case of long names.

But I will add more to the comment as per your suggestion: mentioning that this function will not only fix those long problematic names but short names that PG has autogenerated as well. :+1: